libs/core/src/lib/action-sheet/action-sheet.component.ts

```diff
@@ -0,0 +1,214 @@
+
+    /** Whether should be displayed in compact mode **/
+    @Input()
+    compact = false;
+
+    /** Whether should be displayed in mobile mode **/
+    @Input()
+    mobile = false;
+
+    /** Whenever links should be visible **/
+    @Input()
+    open = false;
+
+    /** Whether internal keyboard support should be enabled. It's enabled by default */
+    @Input()
+    keyboardSupport = true;
+
+    /** To allow user to determine what event he wants to trigger the messages to show
+     * Accepts any [HTML DOM Events](https://www.w3schools.com/jsref/dom_obj_event.asp).
+     **/
+    @Input()
+    triggers: string[] = ['click'];
+
+    /** Event thrown, when focus escapes the list */
+    @Output()
+    focusEscapeList = new EventEmitter<FocusEscapeDirection>();
+
+    /** Event thrown, when the action sheet is opened or closed */
+    @Output()
+    readonly openChange: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+    /** @hidden */
+    @ContentChild(ActionSheetBodyComponent)
+    actionSheetBody: ActionSheetBodyComponent;
+
+    /** @hidden */
+    @ContentChild(ActionSheetControlComponent)
+    actionSheetControl: ActionSheetControlComponent;
+
+    /** @hidden */
+    @ContentChildren(ActionSheetItemComponent, { descendants: true })
+    actionSheetItems: QueryList<ActionSheetItemComponent>;
+
+    /** @hidden */
+    @ViewChild('actionSheetBodyTemplate')
+    actionSheetBodyTemplate: TemplateRef<any>;
+
+    /** @hidden */
+    @ViewChild(PopoverComponent)
+    popoverComponent: PopoverComponent;
+
+    /** @hidden */
+    actionSheetMobileDynamic: ComponentRef<ActionSheetMobileComponent>;
+
+    /** @hidden **/
+    private readonly _onDestroy$: Subject<void> = new Subject<void>();
+
+    /** @hidden **/
+    private readonly _onRefresh$: Subject<void> = new Subject<void>();
+
+    constructor(
+        private _elementRef: ElementRef,
+        private _keyboardSupportService: KeyboardSupportService<ActionSheetItemComponent>,
+        @Optional() private _dynamicComponentService: DynamicComponentService
+    ) {}
+
+
+    /** @hidden */
+    ngAfterContentInit(): void {
+        this._initializeChildrenState();
+        this._keyboardSupportService.setKeyboardService(this.actionSheetItems, false);
+        this._listenOnItemsChange();
+        this._actionControlHandle();
+        this._actionItemsHandle();
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        if (this.mobile) {
+            this._setUpMobileMode();
+        }
+        if (this.popoverComponent && this.popoverComponent) {
+            this.popoverComponent.directiveRef.loaded.pipe(takeUntil(this._onDestroy$)).subscribe(() => {
+                setTimeout(() => {
+                    this.setItemActive(0);
+                });
+            });
+        }
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._onDestroy$.next();
+        this._onDestroy$.complete();
+        this.actionSheetControl.clicked.unsubscribe();
```

Comment:
I'm not sure if it's going to work. After checking my other comment about unsubscribing you would be able to remove this line